cplusplus.pri

```diff
@@ -0,0 +1,80 @@
+# Copyright 2005-2016 The Mumble Developers. All rights reserved.
+# Use of this source code is governed by a BSD-style license
+# that can be found in the LICENSE file at the root of the
+# Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+# This .pri file implements modern C++ detection for Mumble.
+#
+# With this file, it's possible to pass one of
+#   c++03, c++11, c++14 or c++1z
+# to qmake via CONFIG when building Mumble, to
+# manually specify the C++ standard the build
+# should target.
+#
+# If no C++ standard is explicitly chosen on
+# the commandline, an appropriate default will
+# be chosen by an auto-detection mechanism.
+# This mechanism looks at QT_CONFIG and automatically
+# adds c++11, c++14 or c++1z to CONFIG. If none of
+# these are available, it adds c++03 to CONFIG.
```

Comment:
We could document why we use c++03 as our baseline. Namely, the GCC issue(?) we experienced (or was it something different?).